tests/js/blocks/content-helper/structure.test.js

```diff
@@ -4,20 +4,227 @@
 import {
 	render,
 	screen,
+	waitFor,
 } from '@testing-library/react';
 import '@testing-library/jest-dom';
 
 /**
  * Internal dependencies.
  */
 import RelatedTopPostList from '../../../../src/blocks/content-helper/components/related-top-post-list';
+import ContentHelperProvider, { RELATED_POSTS_DEFAULT_LIMIT, RELATED_POSTS_DEFAULT_TIME_RANGE } from '../../../../src/blocks/content-helper/content-helper-provider';
 
 describe( 'Content Helper', () => {
-	it( 'should display spinner when starting', () => {
+	test( 'should display spinner when starting', () => {
+		const getRelatedTopPostsMock = jest
+			.spyOn( ContentHelperProvider, 'getRelatedTopPosts' )
+			.mockImplementation( () => Promise.resolve( {} ) );
+
 		render( <RelatedTopPostList /> );
 
-		const spinner = screen.getByRole( 'presentation' );
+		expect( getRelatedTopPostsMock ).toHaveBeenCalled();
+
+		const spinner = getSpinner();
+		expect( spinner ).toBeInTheDocument();
 		expect( spinner ).toBeVisible();
-		expect( spinner ).toHaveClass( 'components-spinner' );
 	} );
+
+	test( 'should show contact us message when parsely site id is not set', async () => {
+		const getRelatedTopPostsMock = jest
+			.spyOn( ContentHelperProvider, 'getRelatedTopPosts' )
+			.mockImplementation( () => Promise.reject( {
+				errors: {
+					parsely_site_id_not_set: 'error msg',
+				},
+			} ) );
+
+		render( <RelatedTopPostList /> );
+		expect( getSpinner() ).toBeInTheDocument();
+
+		await waitFor( () => screen.findByTestId( 'parsely-contact-us' ), { timeout: 3000 } );
+
+		expect( getRelatedTopPostsMock ).toHaveBeenCalled();
```

Comment:
I wanted to use `toHaveBeenCalledTimes` function which also verifies how much time it is called but it seems like mock counts are persisting between each tests so how can we clear these mock counts between tests?

Tried: `clearAllMocks, resetAllMocks, restoreAllMocks` in `breforeEach and afterEach` callbacks but none of them worked

Reply:
Sounds like, because you are `await`ing, the component renders multiple times and the function is called multiple times.

Reply:
Yes it does call multiple times because of the retry logic present in catch part of the API ...... but the counter of calling mocks should reset if I am doing `clearAllMocks, resetAllMocks, restoreAllMocks` at the start of the test but it isn't happening